ext/src/zpages/tracez_processor.cc

```diff
@@ -0,0 +1,37 @@
+#include "opentelemetry/ext/zpages/tracez_processor.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace ext {
+namespace zpages {
+
+  void TracezSpanProcessor::OnStart(opentelemetry::sdk::trace::Recordable &span) noexcept {
+    if (shutdown_signal_received_) return;
+    spans_.running.insert(static_cast<opentelemetry::sdk::trace::SpanData*>(&span));
+  }
+
+  void TracezSpanProcessor::OnEnd(std::unique_ptr<opentelemetry::sdk::trace::Recordable> &&span) noexcept {
+     if (shutdown_signal_received_ || span == nullptr) return;
+     auto completed_span_raw = static_cast<opentelemetry::sdk::trace::SpanData*>(span.release());
+     auto completed_span_it = spans_.running.find(completed_span_raw);
+     if (completed_span_it != spans_.running.end()) {
+       spans_.running.erase(completed_span_it);
```

Comment:
Is there a race condition? E.g. we could end up having a span removed from `running` but not yet pushed to `completed`.

Reply:
It's possible and we're thinking about adding locks if this is an issue (after tests are written), thanks for noting that and I'll make note of that more explicitly in the future!

Reply:
I think at minimum we need to put a TODO comment here.

Reply:
Yes, I agree and will add that!